src/Controller/CocktailController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Controller;
+
+use App\Form\CocktailFormType;
+use App\Entity\CocktailName;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\HttpFoundation\Request;
+
+class CocktailController extends AbstractController
+{
+    #[Route('/home', name: 'home')]
+    public function new(Request $request): Response
+    {
+        $cocktailName = new CocktailName();
+
+        $form = $this->createForm(CocktailFormType::class, $cocktailName);
+        $form->handleRequest($request);
+
+        if($form->isSubmitted() && $form->isValid()){
+
+            $cocktailName = $form->getData();
```

Comment:
Also, overall I see some missing logic in this controller: so if submit is successful, $cocktailName is set to formData, otherwise it's set as CocktailName entity object? I think this would draw some errors on failure.

```suggestion
            if (!$form->isSubmitted() || $form->isValid()) {
                //it's better to not nest if statements, try to get out of code on failure, or on success
                //render form here with some value that would indicate that it's failure
                 return $this->render('cocktail/home.html.twig', [
                     'cocktailForm' => $form->createView(),
                     'cocktailName' => 'Error on submitting value'
               ]);
           }
               
            $cocktailName->setName($form->getData()); // make sure this will be single element as it would cause errors too.
           
           //return $this->render() as success here
```

Reply:
fixed